### PR TITLE
Switch to extended-type-system/type:0.2.x

### PR DESCRIPTION
**Pull request by aivchen**

### CODE REVIEW

File: src/TypeStringifier.php [hunk 1]
```diff
@@ -129,118 +151,133 @@ public function visitIntRange(IntRangeT $type): mixed
...
+
+    public function visitScalar(ScalarType $type): mixed
     {
-        if ($type->valueType instanceof MixedT) {
+        return 'scalar';
+    }
+
+    public function visitNonEmptyList(NonEmptyListType $type): mixed
+    {
+        if ($type->valueType instanceof MixedType) {
```
Comment (vudaltsov) [inline src/TypeStringifier.php:241]: `$type->valueType === types::mixed` and all other atomic type comparisons below (`array-key`).

**Review by vudaltsov — CHANGES_REQUESTED** [review]

**Comment by vudaltsov** [comment]
Change the PR name, please.

**Comment by vudaltsov** [comment]
And it seems that locally you've committed directly to `0.2.x` instead of creating a separate branch. Fix that, otherwise you will have problems with pulling later. If you have problems with that, let me know.

File: src/TypeStringifier.php (hunk 1 above)
Comment (aivchen) [reply src/TypeStringifier.php:241]: In this case we get a psalm error:
```bash
Type ExtendedTypeSystem\Type<mixed> for $type->valueType is never =enum(ExtendedTypeSystem\Type\MixedType::self) (see https://psalm.dev/056)
        if ($type->valueType === types::mixed) {
```

Comment (vudaltsov) [reply src/TypeStringifier.php:241]: Ok!

**Review by vudaltsov — APPROVED** [review]

**Comment by vudaltsov** [comment]
Thank you, @aivchen!

**Comment by Lucky4ever6129** [comment]
Hi, @aivchen.
Did you checked my code?

**Pull request merged**